src/cli/add.rs

```diff
@@ -160,14 +171,47 @@ pub async fn add_specs_to_project(
     }
 
     // Update the lock file and write to disk
-    update_lock_file(
+    let lock_file = update_lock_file(
         project,
         load_lock_file(project).await?,
         Some(sparse_repo_data),
     )
     .await?;
     project.save()?;
 
+    if !no_install {
+        let platform = Platform::current();
+        if project.platforms().contains(&platform) {
+            // Get the currently installed packages
+            let prefix = Prefix::new(project.root().join(".pixi/env"))?;
+            let installed_packages = prefix.find_installed_packages(None).await?;
+
+            // Construct a transaction to bring the environment up to date with the lock-file content
+            let transaction = Transaction::from_current_and_desired(
+                installed_packages,
+                get_required_packages(&lock_file, platform)?,
+                platform,
+            )?;
+
+            // Execute the transaction if there is work to do
+            if !transaction.operations.is_empty() {
+                // Execute the operations that are returned by the solver.
+                await_in_progress(
+                    "updating environment",
+                    execute_transaction(
+                        transaction,
+                        prefix.root().to_path_buf(),
+                        rattler::default_cache_dir()?,
+                        AuthenticatedClient::default(),
+                    ),
+                )
+                .await?;
```

Comment:
For me this is big enough of a code duplication in between this and `get_up_to_date_prefix` to warrant getting it into a separate function.